### ISSUE TITLE
Feature Request: Provide optional in-app check for updated ZIM archives

### ISSUE THREAD

**Issue by jojo2357** [kiwix/kiwix-js-windows#385]
I currently have an archive of Wikipedia, and I noticed that there is currently a newer once available for download. Is there/can there be a way to update/replace outdated archives without just deleting the old one and downloading the new one by hand? 

It would be best if unneeded data could be omitted from the re-download--essentially just redoing the changed and new stuff--but if this is not feasible, at least a way to automate/automagically update archives.

Part of my issue is that my archives went out of date and I don't know which ones have newer versions and which do not. automation is also of the essence

TL;DR

Automatically update/show when there are new versions of an archive.

**Comment by Jaifroid** [comment]
@jojo2357 For those using Chromium browsers or a framework like UWP or Electron, It would be possible to have an online check run against the list of archives and change their colour (or flag in some other way) those that are outdated.

Updating parts of an archive is a much-requested feature but not technically possible currently, because of the highly compressed file format. Information isn't stored in predictable packets that can be marked as updated or not. While it *might* be possible to do some byte-level file differences between the new archive and a local one, my guess is that almost everything would be different and/or in different positions in the file, and it is much easier just to download the new file and archive (or delete) the old one.

Somewhat related to this issue is #287, which would be technically possible, but is, frankly, unlikely to be implemented because it would be technically pretty complex and would make the app slow.

TL;DR - I could implement a check to indicate that a newer version of an archive is available (when the app is online), but it would not be possible to update it automatically: the user would still need to initiate the download of the complete new archive.

**Comment by jojo2357** [comment]
> https://www.reddit.com/r/Kiwix/comments/x2wkbm/kiwixzim_a_bash_script_to_check_downloadkiwixorg/

Did exactly as I wanted (just realized this kiwix is for windows; I have linux so the script worked right out of the box).

While I would like to see this in the app, I will not be upset if this gets closed.

**Comment by Jaifroid** [comment]
That's good to know! Actually, the name of this Repo is misleading: the app is for Windows and Linux (we release Linux Electron packages of the app), and the PWA is universal (Linux, Windows, iOS, macOS), but because it started as a Windows port, the name has stuck.

I do aim to provide an optional check for updates to archives in-app, but it may take a few months! I'll change the title of this to reflect the aim.
